app/models/document.rb

```diff
@@ -1,6 +1,6 @@
 class Document < ApplicationRecord
   belongs_to :company
   belongs_to :currency
-  belongs_to :financial_institution
   belongs_to :document_type
+  has_one :financial_institution
```

Comment:
are you sure?

Reply:
I think it is correct to say that a document has (or does not have) an associated financial institution. for example:
A "Cheque" type document has a financial institution associated with it, but a "Pagare" does not!

Reply:
No, it's not. has_one is a special case of has_many, basically has_one is equals to has_many but limited to 1 record
The question is where the foreign key goes
If you specify has_one in Document model, it is expected to have a `document_id` into `finantial_institutions` table, which is not the case
It should be belongs_to :financial_institution, optional: true